### ISSUE TITLE
noise output av model posttrain

### ISSUE THREAD

**Issue by dwanfun** [nvidia-cosmos/cosmos-transfer1#77]
After AV mode hdmap post-training, the model inference output is pure noise. I suspect there is a problem with the conversion of ckpts, here is the output video and convert logs. please take a look

[convert.log](https://github.com/user-attachments/files/20075017/convert.log)

https://github.com/user-attachments/assets/02b5669d-e5a1-4aba-a477-1bb10d2c1c6d

**Comment by YirunLi** [comment]
I had the same problem. After training Cosmos-Transfer1-7B, I merged the models and the inference results were as follows.
https:github.com/user-attachments/assets/7995090e-8a5b-4bd3-bc6c-1d24524488d7
Are there detailed instructions for using the new model for reasoning after post-training, or other ways to verify whether the training results and the dataset are correct?

**Comment by arieling** [comment]
@dwanfun can you tell me more on hot to reproeduce this error?
What did you do exactly and the commands

**Comment by YirunLi** [comment]
I used depth control to post-train cosmos-transfer by inputting depth video.

Firstly, according to https:Following the instructions of github.com/nvidia-cosmos/cosmos-transfer1/blob/main/examples/training_cosmos_transfer_7b.md, the data was preprocessed, T5 Text Embeddings were generated, and base_model.pt and depth_control.pt were split using the scripts/convert_ckpt_fsdp_to_tp.py script.

Then, 175,000 iter were trained on 8 x H100s. After training, the tp model was merged into iter_000175000_reg_model.pt and iter_000175000_ema_model_only.pt using the convert_ckpt_tp_to_fsdp.py script.

According to the description in the paper, base_model is frozen during training, so I only replaced the model of Cosmos-Transfer1-7B/depth_control.pt with iter_000175000_ema_model_only.pt that I just merged, and this result appeared. 

Is there any problem with the whole process?

> I had the same problem. After training Cosmos-Transfer1-7B, I merged the models and the inference results were as follows.
>
> ![Image](https:github.com/user-attachments/assets/24140d4b-c6d6-4369-b304-ce8f0c2b129c)
>
> Are there detailed instructions for using the new model for reasoning after post-training, or other ways to verify whether the training results and the dataset are correct?

**Comment by caotians1** [comment]
@dwanfun and @YirunLi Does this problem occur if you only train for very few iterations (e.g. 10 iters)? Can you provide a few more details regarding your setup? Specifically:
1. are you using the conda environment or the docker container?
2. was inference ran with single GPU or multi GPU?
3. @YirunLi can you also capture the output when running convert_ckpt_tp_to_fsdp.py and share it here?

**Comment by dwanfun** [comment]
Hi, I'm using a conda environment and running inference on 8 GPUs.
I have re-converted the FSDP to TP weights via convert_ckpt_fsdp_to_tp, and the model can now be trained normally.  
However, the loss behaves differently compared to before — previously it was greater than 1, and sometimes even reached 10. Now it stays around 0.x and occasionally becomes negative.
I was wondering if this is expected, and whether such loss values are considered normal during training?
 I encountered another issue while running the convert_ckpt_tp_to_fsdp script, some layers have shape mismatches.
Thanks in advance!

[convert_new.log](https://github.com/user-attachments/files/20119214/convert_new.log)

**Comment by YirunLi** [comment]
Thank you for your reply.
I tried training with 10 iters as you suggested, and the results were surprisingly good, it seems I used too many iters for training before. 
Origin video captrue:
![Image](https://github.com/user-attachments/assets/18d8bd9e-2d7c-412a-bb24-f3b623bbc82b)
Output video capture:
![Image](https://github.com/user-attachments/assets/5d3d7795-38a1-4b04-b93e-27356b1bddaa)

I previously only prepared a small amount of training data to test the training pipeline, so it's possible that too many iters led to training failure.
Then, in addition to testing the generation effect of each saved model, are there other methods to determine how many iters should be trained?

**Comment by caotians1** [comment]
Hi @dwanfun , the new conversion logs looks correct - we expect missing keys in the "load_base_model" function since the base model weights are not stored in the ctrlnet checkpoints. It is great that you are observing close to zero or negative loss values - this is what we expect when finetuning the model on data of similar domain (incl. waymo).

**Comment by caotians1** [comment]
@YirunLi Thanks for posting the new results! It is great that the checkpoint conversion and inference scripts are now working as intended. If you are still encountering training instability issues, please create a new issue with information of the base checkpoint, post-training config, and dataset metadata so that we can better support your use-case.

**Comment by dwanfun** [comment]
@caotians1  I still suspect there's an issue with the weight conversion, as there are not only missing keys but also some layer shape mismatches.

![Image](https://github.com/user-attachments/assets/12fcc148-7174-4fca-918e-923d0031912a)

**Comment by qianlim** [comment]
> Can anyone tell me how long it takes to run for 10k iterations on 8x H100 or A100? I am trying to understand the cost to fine-tuning the model. Any information is much appreciated.

@lbferreira For 7B model, the iteration speed on 8xH100 is roughly 4-5s/iter.

**Comment by lbferreira** [comment]
@qianlim 
Thank you for your response.
